### PR TITLE
change to single docker file for multiple arch's

### PR DESCRIPTION
**Pull request by garrygerber**
Dockerfile now supports all arch's, control on the arch is made by the variable "TARGETPLATFORM"

using the following plugin [drone-docker-buildx](https://drone-plugin-index.geekdocs.de/plugins/drone-docker-buildx/) which uses the "docker buildx" command, you can set a flag "platforms" to the plugin and pass with comma-separation, list of the arch's you want
I tried to update the drone.yml but can't test it, hope this is enough to allow you to fix issue #533

### CODE REVIEW

**Comment by oliver006** [comment]
Thanks for the PR - I'll try to get to reviewing it as soon as I can, hopefully in the next 1-2 days.

**Comment by oliver006** [comment]
Sorry it took me a while to get to this.
The general approach makes sense, two things we should look into:
- something is wrong with your yaml file, you can see the error here: https://cloud.drone.io/oliver006/redis_exporter/1244 - I think line 221 is not indented right
- I'd like to avoid using a 3rd party plugin from an unknown contributor (thegeeklab/drone-docker-buildx). Let me look into that plugin and what it does differently compared to the official plugin and maybe there's an alternate way to make this work with the official plugin.

**Comment by oliver006** [comment]
I think the issue is that the docker build step needs to run in privileged mode. Can you try adding an attribute `privileged: true` and see if that helps? It might not be possible to run the geeklabs image in privileged mode, in that case you could try using the regular docker image and directly invoke the build command via the `commands` parameter.

**Comment by garrygerber** [comment]
this doesn't work with the plugins/docker image.

another way:
can you add to the DRONE_RUNNER_PRIVILEGED_IMAGES env var - "thegeeklab/drone-docker-buildx"
according to https://docs.drone.io/runner/docker/configuration/reference/drone-runner-privileged-images/
you can set the default containers that start with privileged mode

**Comment by oliver006** [comment]
Unfortunately  I don't own or control the cloud.drone.io instance and don't have a way to set the `DRONE_RUNNER_PRIVILEGED_IMAGES` variable.

**Comment by garrygerber** [comment]
using the docker_pass and docker_user helped, it passed and built the manifest!
its working and ready for merge, looking forward to your review
Thanks!

**Comment by oliver006** [comment]
Thanks for figuring this out, great work!
I'll try to review as soon as possible, hopefully in the next few days.

File: .drone.yml [hunk 1]
```diff
@@ -313,24 +331,51 @@ steps:
...
-      dockerfile: ./docker/Dockerfile.amd64
+      dockerfile: ./docker/Dockerfile
       target: scratch
       build_args:
         - 'TAG=${DRONE_TAG}'
         - 'SHA1=${DRONE_COMMIT_SHA}'
         - 'GOARCH=amd64'
       config:
         from_secret: docker_config_json
+      username:
```
Comment (oliver006) [inline .drone.yml:375]: This shouldn't be necessary for quay, they're for docker hub only..

File: .drone.yml [hunk 2]
```diff
@@ -340,18 +385,23 @@ steps:
...
+      tags: "latest-arm64,${DRONE_TAG}-arm64,alpine-arm64"
+      dockerfile: ./docker/Dockerfile
+      target: alpine
       build_args:
         - 'TAG=${DRONE_TAG}'
         - 'SHA1=${DRONE_COMMIT_SHA}'
         - 'GOARCH=arm64'
       config:
         from_secret: docker_config_json
+      username:
```
Comment (oliver006) [inline .drone.yml:401]: Same

File: .drone.yml [hunk 3]
```diff
@@ -361,18 +411,23 @@ steps:
...
+      tags: "latest-arm,${DRONE_TAG}-arm,alpine-arm"
+      dockerfile: ./docker/Dockerfile
+      target: alpine
       build_args:
         - 'TAG=${DRONE_TAG}'
         - 'SHA1=${DRONE_COMMIT_SHA}'
         - 'GOARCH=arm'
       config:
         from_secret: docker_config_json
+      username:
```
Comment (oliver006) [inline .drone.yml:427]: Same

File: .drone.yml [hunk 4]
```diff
@@ -382,27 +437,56 @@ steps:
...
+      tags: "${DRONE_TAG}-alpine,alpine-amd64"
+      dockerfile: ./docker/Dockerfile
       target: alpine
       build_args:
         - 'TAG=${DRONE_TAG}'
         - 'SHA1=${DRONE_COMMIT_SHA}'
         - 'GOARCH=amd64'
       config:
         from_secret: docker_config_json
+      username:
```
Comment (oliver006) [inline .drone.yml:453]: Same

File: .drone.yml [hunk 5]
```diff
@@ -382,27 +437,56 @@ steps:
...
         - tag
     depends_on:
       - tests
 
+
+  - name: manifest-quay-alpine
+    image: plugins/manifest
+    pull: if-not-exists
+    settings:
+      username:
```
Comment (oliver006) [inline .drone.yml:468]: This should use the json config, not the docker username and password.

**Review by oliver006 — COMMENTED** [review]
Left a few comments.
Also, something is trimming the end-of-file line feeds of files you touch - maybe you can revert that and turn that off.
Thanks!

**Comment by garrygerber** [comment]
@oliver006 Thanks for the review, I've fixed all I think
can you review it again?

**Comment by oliver006** [comment]
Thanks @garrygerber for seeing this through to the end - I think this looks good, let's hope it all works once I merge and tag a release!

**Comment by garrygerber** [comment]
Thanks @oliver006 glad we made it work.
looking forward to your tag and release!

**Comment by oliver006** [comment]
Had to push one minor tweak but I think it worked!
Released as v1.31.0

**Pull request merged**